octopoes/octopoes/repositories/ooi_repository.py

```diff
@@ -296,9 +305,21 @@ def list_oois(
         limit: int = 20,
         scan_levels: set[ScanLevel] = DEFAULT_SCAN_LEVEL_FILTER,
         scan_profile_types: set[ScanProfileType] = DEFAULT_SCAN_PROFILE_TYPE_FILTER,
+        search_string: str | None = None,
+        order_by: Literal["scan_level", "object_type"] = "object_type",
+        asc_desc: Literal["asc", "desc"] = "asc",
     ) -> Paginated[OOI]:
         types = to_concrete(types)
 
+        search_statement = (
+            f"""[?e :xt/id ?id]
+                                [(clojure.string/includes? ?id \"{escape_string(search_string)}\")]"""
+            if search_string
+            else ""
+        )
+
+        order_statement = f":order-by [[_{re.escape(order_by)} :{re.escape(asc_desc)}]]"
```

Comment:
Order_by and asc_desc can only be the literal strings specified in the FastAPI api method, so there is no need to escape them. And if they could be arbitrary strings the regular expressions escaping would also not escape all the datalog special characters as far as I can see.

```suggestion
        order_statement = f":order-by [[_{order_by} :{asc_desc}]]"
```